File: 프로그래머스/1/161990. 바탕화면 정리/바탕화면 정리.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<string> wallpaper) {
    vector<int> answer;
    
    int min_x = 1e9, min_y = 1e9;
    int max_x = -1, max_y = -1;
    
    for(int i = 0 ; i < wallpaper.size() ; i++)
        for(int j = 0 ; j < wallpaper[i].size(); j++)
            if(wallpaper[i][j] == '#'){
                min_x = min(min_x, j);
                max_x = max(max_x, j);
                min_y = min(min_y, i);
                max_y = max(max_y ,i);
            }

    answer.push_back(min_y);
    answer.push_back(min_x);
    answer.push_back(max_y + 1);
    answer.push_back(max_x + 1);
    
    return answer;
}

File: 프로그래머스/1/161990. 바탕화면 정리/바탕화면 정리.java
class Solution {
    public int[] solution(String[] wallpaper) {
        
        int minY = Integer.MAX_VALUE;
        int minX = Integer.MAX_VALUE;
        
        int maxY = Integer.MIN_VALUE;
        int maxX = Integer.MIN_VALUE;
        
        for(int i = 0 ; i < wallpaper.length ; i++)
            for(int j = 0 ; j < wallpaper[i].length() ; j++)
                if(wallpaper[i].charAt(j) == '#'){
                    minY = Math.min(minY, i);
                    minX = Math.min(minX, j);
                    maxY = Math.max(maxY, i);
                    maxX = Math.max(maxX, j);
                }
        
        return new int[]{minY, minX, maxY + 1, maxX + 1};
        
    }
}